test/integration/src/py/conftest.py

```diff
@@ -33,14 +42,43 @@ def ethereum_network():
 
 
 @pytest.fixture
-def sifnodecli_homedir():
-    return f"""{test_utilities.get_required_env_var("CHAINDIR")}/.sifnodecli"""
+def solidity_json_path(smart_contracts_dir):
+    return test_utilities.get_optional_env_var("SOLIDITY_JSON_PATH", f"{smart_contracts_dir}/build/contracts")
 
 
 @pytest.fixture
-def rowan_source(is_ropsten_testnet):
+def sifnodecli_homedir(is_ropsten_testnet):
+    if is_ropsten_testnet:
+        base = test_utilities.get_required_env_var("HOME")
+    else:
+        base = test_utilities.get_required_env_var("CHAINDIR")
+    result = f"""{base}/.sifnodecli"""
+    return result
+
+
+@pytest.fixture
+def rowan_source(is_ropsten_testnet, validator_address):
     """A sifchain address or key that has rowan and can send that rowan to other address"""
-    return test_utilities.get_required_env_var("ROWAN_SOURCE")
+    result = test_utilities.get_optional_env_var("ROWAN_SOURCE", None)
+    if result:
+        return result
+    if is_ropsten_testnet:
+        assert result
+    else:
+        assert validator_address
+        return validator_address
+
+
+@pytest.fixture
+def rowan_source_key(is_ropsten_testnet):
+    """A sifchain address or key that has rowan and can send that rowan to other address"""
+    result = test_utilities.get_optional_env_var("ROWAN_SOURCE_KEY", None)
+    if result:
+        return result
+    if is_ropsten_testnet:
+        assert result
```

Comment:
We don't need to return this?

Reply:
No, this just asserts that you've set it manually.  Added a comment.